can/exceptions.py

```diff
@@ -0,0 +1,61 @@
+"""
+There are several specific :class:`Exception` classes to allow user
+code to react to specific scenarios related to CAN busses::
+
+    Exception (Python standard library)
+     +-- ...
+     +-- CanError (python-can)
+         +-- CanBackendError
+         +-- CanInitializationError
+         +-- CanOperationError
+         +-- CanTimeoutError
+
+Keep in mind that some functions and methods may raise different exceptions.
+For example, validating typical arguments and parameters might result in a
+:class:`ValueError`. This should be documented for the function at hand.
+"""
+
+
+class CanError(Exception):
+    """Base class for all CAN related exceptions."""
+
+
+class CanBackEndError(CanError):

```

Comment:
Perhaps the name `CanInterfaceError` lines up better with the code?

Consider adding an optional argument with the interface name.

Reply:
The concrete interface being used should probably always be clear from both the context in which it was raised (most projects probably only use very few interfaces) and the stack trace. Would you still say it would be useful?

Reply:
Yes, I think that name would be a better fit, but also not perfect. Maybe, we don't really need it at all? If we have a separate error for the interface not being found, this error might become unnecessary? Because the new `CanInterfaceNotImplemented` could also encompass drivers not being present. As a user, this seems like a reasonable error: The functionality the user seeks to use is not implemented on the current system. Also, I think that the interface not being usable since it never exists and it not being available due to missing drivers is something that should be clear from the message. Having separate classes might make stuff more complicated than it needs to be.

Reply:
💯  Yes agree completely that if we can reduce the number of exceptions a user needs to be aware of that is a win.

You're right, no need to add complexity with the interface name as this (or rather the `CanInterfaceNotImplemented` exception will always be coming from a particular interface (or explicitly mentioned in the exception message).